docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,346 @@
+3. Any storage capacity used for PDBs and source should not impact this feature, such as compression algorithm.
+4. Only Portable PDB files will be included for this spec. We can expand this feature past these once it is implemented and proven needed elsewhere.
+
+This document will provide the expanded specification to the Portable PDB format. Any additions to that format will be ported to expand documentation provided in [dotnet-runtime](https://github.com/jnm2/dotnet-runtime/blob/26efe3467741fe2a85780b2d2cd18875af6ebd98/docs/design/specs/PortablePdb-Metadata.md#source-link-c-and-vb-compilers).
+
+## List of Compiler Flags
+
+#### CSharp Flags That Can Be Derived From PDB or Assembly
+
+* baseaddress
+* checksumalgorithm
+* debug
+* deterministic
+* embed
+* filealign
+* highentropyva
+* link
+    - Represented by a metadata reference with `EmbededInteropTypes=true` 
+* linkresource
+    - Will be represented in metadata reference embedded in pdb
+* main
+    - Already stored in PDB as the entry point token
+* moduleassemblyname
+* modulename
+* nostdlib
+    - Will be represented in metadata reference embedded in pdb
+* nowin32manifest
+* pdb
+* platform
+* publicsign
+* resource
+    - Will be represented in metadata reference embedded in pdb
+* subsystemversion
+* target
+* win32icon
+* win32manifest
+* win32res
+
+#### CSharp Flags Not Included
+
+* bugreport
+* delaysign
+* doc
+* errorendlocation
+* errorlog
+* errorreport
+* fullpaths
+* incremental
+* keycontainer
+* keyfile
+* noconfig
+* nologo
+* nowarn
+* out
+* parallel
+* pathmap
+* preferreduilang
+* recurse
+* refout
+* refonly
+* reportanalyzer
+* ruleset
+* utf8output
+* version
+* warn
+* warnaserror
+
+#### Visual Basic Flags That Can Be Derived From PDB or Assembly
+
+* baseaddress
+* checksumalgorithm
+* debug
+* filealign
+* highentropyva
+* linkresource
+    - Will be represented in metadata reference embedded in pdb
+* main 
+* nostdlib
+    - Will be represented in metadata reference embedded in pdb
+* platform
+* resource
+    - Will be represented in metadata reference embedded in pdb
+* subsystemversion
+* target
+* win32icon
+* win32manifest
+* win32resource
+
+#### Visual Basic Flags Not Included
+
+* bugreport
+* delaysign
+* doc
+* errorreport
+* help
+* keycontainer
+* keyfile
+* libpath
+* moduleassemblyname
+* modulename
+* netcf
+* noconfig
+* nologo
+* nowarn
+* nowin32manifest
+* optimize
+* optioncompare
+* optionexplicit
+* optioninfer
+* out
+* parallel
+* preferreduilang
+* quiet
+* recurse
+* refonly
+* refout
+* rootnamespace
+* ruleset
+* sdkpath
+* utf8output
+* vbruntime
+* verbose
+* warnaserror
+
+#### Compiler Version
+
+This will be a full compiler version, including commit SHA for the build of the compiler. This will be used since compilations across compiler versions are not guaranteed to be the same.
+
+#### Source Encoding
+
+Encoding for source files that are not declared using BOM. This 
```

Comment:
Need to finish out this paragraph. Presently ends with an incomplete sentence.

Reply:
woops 😑 adding more info about how this is retrieved and stored